internal/security/secretstore/init.go

```diff
@@ -159,6 +172,32 @@ func (b *Bootstrap) Handler(
 	gk := NewGokeyGenerator(absPath)
 	loggingClient.Warn("WARNING: The gokey generator is a reference implementation for credential generation and the underlying libraries not been reviewed for cryptographic security. The user is encouraged to perform their own security investigation before deployment.")
 	cred := NewCred(req, absPath, gk, configuration.SecretService.GetSecretSvcBaseURL(), loggingClient)
+
+	// Detour: Create secrets engine if not there already
+	vaultToken, err := GetAccessToken(absPath)
+	if err != nil {
+		loggingClient.Error(fmt.Sprintf("failed to reload root token: %s", err.Error()))
+		os.Exit(1)
+	}
+	// Note: trailing slash on secret/ is required.
+	installed, err := vc.CheckSecretEngineInstalled(vaultToken, "secret/", "kv")
+	if err != nil {
+		loggingClient.Error(fmt.Sprintf("failed call to check if kv secrets engine is insalled: %s", err.Error()))
+		os.Exit(1)
+	}
+	if !installed {
+		loggingClient.Info("enabling kv secrets engine for the first time...")
+		// Enable KV version 1 at /v1/secret path (/v1 prefix supplied by Vault)
+		_, err := vc.EnableKVSecretEngine(vaultToken, "secret", "1")
+		if err != nil {
+			loggingClient.Error(fmt.Sprintf("failed call to enable KV secret store: %s", err.Error()))
+			os.Exit(1)
+		}
+	} else {
+		loggingClient.Info("kv secrets engine already enabled...")
```

Comment:
ditto

Reply:
Changed to "KV secrets engine" everywhere.